server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1901,18 +2026,21 @@ private Set<DataSegment> createNewIdsOfAppendSegmentsAfterReplace(
                                                 .findFirst().orElse(null);
     final Map<String, String> upgradeSegmentToLockVersion
         = getAppendSegmentsCommittedDuringTask(handle, taskId);
-    final List<DataSegment> segmentsToUpgrade
+
+    final List<DataSegmentWithSchemaInformation> segmentsToUpgrade
         = retrieveSegmentsById(handle, datasource, upgradeSegmentToLockVersion.keySet());
 
+    log.info("createNewIdsOfAppendSegmentsAfterReplace SegmentsToUpgrade are [%s]", segmentsToUpgrade);
```

Comment:
Not needed. Whenever adding logs, please make sure that they are really adding meaningful information that can be used for debugging scenarios.